### PR TITLE
Additional URIPathCanonicalization Tests

### PR DESCRIPTION
**Pull request by pmd1nh**
for #20414

Add tests for skipping encoded char verification via web.xml and server.xml

### CODE REVIEW

**Comment by pmd1nh** [comment]
#build 
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat

File: dev/io.openliberty.webcontainer.servlet.6.0.internal_fat/fat/src/io/openliberty/webcontainer/servlet60/fat/tests/Servlet60URIPathCanonicalizationServerXMLTest.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+    @Server("servlet60_uriPathCanonicalServerXMLTest")
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        // Create the WAR and add resources
+        WebArchive uriPathCanonicalizeWar = ShrinkWrap.create(WebArchive.class, WAR_NAME);
+        ShrinkHelper.addDirectory(uriPathCanonicalizeWar, "test-applications/" + WAR_NAME + "/resources");
+        ShrinkHelper.defaultDropinApp(server, WAR_NAME, "servlets");
```
Comment (pnicolucci) [inline dev/io.openliberty.webcontainer.servlet.6.0.internal_fat/fat/src/io/openliberty/webcontainer/servlet60/fat/tests/Servlet60URIPathCanonicalizationServerXMLTest.java]: I don't think you need the first two lines here do you?

File: dev/io.openliberty.webcontainer.servlet.6.0.internal_fat/fat/src/io/openliberty/webcontainer/servlet60/fat/tests/Servlet60URIPathCanonicalizationWebXMLTest.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    private static final String WAR_NAME = TEST_APP_NAME + ".war";
+
+    @Server("servlet60_uriPathCanonicalizationTest")
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        // Create the WAR and add resources
+        WebArchive uriPathCanonicalizeWar = ShrinkWrap.create(WebArchive.class, WAR_NAME);
+        ShrinkHelper.addDirectory(uriPathCanonicalizeWar, "test-applications/" + WAR_NAME + "/resources");
```
Comment (pnicolucci) [inline dev/io.openliberty.webcontainer.servlet.6.0.internal_fat/fat/src/io/openliberty/webcontainer/servlet60/fat/tests/Servlet60URIPathCanonicalizationWebXMLTest.java]: Are the first two lines necessary?

File: dev/io.openliberty.webcontainer.servlet.6.0.internal_fat/fat/src/io/openliberty/webcontainer/servlet60/fat/tests/Servlet60URIPathCanonicalizationWebXMLTest.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+ * should NOT be rejected.
+ */
+@RunWith(FATRunner.class)
+@Mode(TestMode.FULL)
+public class Servlet60URIPathCanonicalizationWebXMLTest {
+    private static final Logger LOG = Logger.getLogger(Servlet60URIPathCanonicalizationWebXMLTest.class.getName());
+    private static final String TEST_APP_NAME = "URIPathCanonicalWebXmlTest";
+    private static final String WAR_NAME = TEST_APP_NAME + ".war";
+
+    @Server("servlet60_uriPathCanonicalizationTest")
```
Comment (pnicolucci) [inline dev/io.openliberty.webcontainer.servlet.6.0.internal_fat/fat/src/io/openliberty/webcontainer/servlet60/fat/tests/Servlet60URIPathCanonicalizationWebXMLTest.java]: Lets create a new server for this test suite as well.

**Comment by pmd1nh** [comment]
cancelled previous PB

**Comment by pmd1nh** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat

**Comment by pmd1nh** [comment]
@pnicolucci can  you review this one?  It is ready to merge.  Thanks!

**Comment by pmd1nh** [comment]
Restructure the test applications to distinguish the servlet paths

#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat

**Review by pnicolucci — APPROVED** [review]

**Comment by pmd1nh** [comment]
#libby

**Pull request merged**